test/moves/magic_coat.test.ts

```diff
@@ -0,0 +1,353 @@
+import type { EncoreTag } from "#app/data/battler-tags";
+import { allAbilities, allMoves } from "#app/data/data-lists";
+import { Abilities } from "#enums/abilities";
+import { ArenaTagSide } from "#enums/arena-tag-side";
+import { ArenaTagType } from "#enums/arena-tag-type";
+import { BattlerIndex } from "#enums/battler-index";
+import { BattlerTagType } from "#enums/battler-tag-type";
+import { MoveId } from "#enums/move-id";
+import { MoveResult } from "#enums/move-result";
+import { Species } from "#enums/species";
+import { Stat } from "#enums/stat";
+import { StatusEffect } from "#enums/status-effect";
+import { GameManager } from "#test/testUtils/gameManager";
```

Comment:
```suggestion
import { GameManager } from "#test/test-utils/gameManager";
```
this will need updating with the latest merges to beta